src/overlays/gamestates/ovl_file_choose/z_file_choose.c

```diff
@@ -1,37 +1,1886 @@
+    this->titleAlpha[0] += VREG(1);
+    this->windowAlpha += VREG(2);
+
+    for (i = 0; i < 3; i++) {
+        this->fileButtonAlpha[i] = this->windowAlpha;
+
+        if (SLOT_OCCUPIED(sramCtx, i)) {
+            this->nameBoxAlpha[i] = this->nameAlpha[i] = this->windowAlpha;
+            this->connectorAlpha[i] += VREG(1);
+            if (this->connectorAlpha[i] >= 255) {
+                this->connectorAlpha[i] = 255;
+            }
+        }
+    }
+
+    this->actionButtonAlpha[BTN_ACTION_COPY] = this->actionButtonAlpha[BTN_ACTION_ERASE] = this->optionButtonAlpha =
+        this->windowAlpha;
+}
+
+/**
+ * Converts a numerical value to ones-tens-hundreds digits
+ */
+void FileChoose_SplitNumber(u16 value, s16* hundreds, s16* tens, s16* ones) {
+    *hundreds = 0;
+    *tens = 0;
+    *ones = value;
+
+    while (true) {
+        if ((*ones - 100) < 0) {
+            break;
+        }
+        (*hundreds)++;
+        *ones -= 100;
+    }
+
+    while (true) {
+        if ((*ones - 10) < 0) {
+            break;
+        }
+        (*tens)++;
+        *ones -= 10;
+    }
+}
+
+/**
+ * Reduce the alpha of the black screen fill to create a fade in effect.
+ * Additionally, slide the window from the right to the center of the screen.
+ * Update function for `CM_FADE_IN_START`
+ */
+void FileChoose_StartFadeIn(GameState* thisx) {
+    FileChooseContext* this = (FileChooseContext*)thisx;
+
+    FileChoose_FadeInMenuElements(&this->state);
+    sScreenFillAlpha -= 40;
+    this->windowPosX -= 20;
+
+    if (this->windowPosX <= -94) {
+        this->windowPosX = -94;
+        this->configMode = CM_FADE_IN_END;
+        sScreenFillAlpha = 0;
+    }
+}
+
+/**
+ * Finish fading in the remaining menu elements.
+ * Fade in the controls text at the bottom of the screen.
+ * Update function for `CM_FADE_IN_END`
+ */
+void FileChoose_FinishFadeIn(GameState* thisx) {
+    FileChooseContext* this = (FileChooseContext*)thisx;
+
+    this->controlsAlpha += VREG(1);
+    FileChoose_FadeInMenuElements(&this->state);
+
+    if (this->titleAlpha[0] >= 255) {
+        this->titleAlpha[0] = 255;
+        this->controlsAlpha = 255;
+        this->windowAlpha = 200;
+        this->configMode = CM_MAIN_MENU;
+    }
+}
+
+/**
+ * Update the cursor and wait for the player to select a button to change menus accordingly.
+ * If an empty file is selected, enter the name entry config mode.
+ * If an occupied file is selected, enter the `Select` menu mode.
+ * If copy, erase, or options is selected, set config mode accordingly.
+ * Lastly, set any warning labels if appropriate.
+ * Update function for `CM_MAIN_MENU`
+ */
+void FileChoose_UpdateMainMenu(GameState* thisx) {
+    static u8 gEmptyName[] = { 0x3E, 0x3E, 0x3E, 0x3E, 0x3E, 0x3E, 0x3E, 0x3E };
+    FileChooseContext* this = (FileChooseContext*)thisx;
+    SramContext* sramCtx = &this->sramCtx;
+    Input* controller1 = &this->state.input[0];
```

Comment:
fwiw there are a few other uses of controller in the codebase. but yeah I can see how input makes more sense.

I think it should be standard to include the number for which controller the input came from though. It only really matters for debug rom, but it can be pretty confusing to see code out of context that is using "input" and its not from controller1/index 0

Reply:
To me it doesn't seem necessary to include which controller is being used unless there are multiple ones being used in the same function or something similar. And I don't know about using `input1` for `input[0]` (or `input3` for `input[2]` for instance), since that sounds like it would make things more confusing overall.

Maybe instead we should use names like `debugInput` when the input isn't from the normal controller, and leave it to the reader to check which extra controller it corresponds to. This would also allow people to switch the code to another controller without having to rename entire variables for it or ending up with inaccurate names.